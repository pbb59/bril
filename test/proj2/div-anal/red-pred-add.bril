# ARGS: diverge

main {
init:
  zero: int = const 0;
  one: int = const 1;
  two: int = const 2;
  three: int = const 3;
pack:
  a0: int = const 7;
  a1: int = const 11;
  vec0: vector = s2vb a0;
  vec1: vector = s2vb a1;
  vec2: vector = s2v zero one two three;
  vec3: vector = s2v one zero two three;
  p0: pred = vcmp vec2 vec3;
add0:
  (p0) vec4: vector = vadd vec0 vec0;
add1:
  (!p0) vec5: vector = vadd vec1 vec1;
phi:
  vec6: vector = vphi p0 vec4 vec5;
extract:
  a2: int = v2s vec6 zero;
  a3: int = v2s vec6 one;
  a4: int = v2s vec6 two;
  a5: int = v2s vec6 three;
print:
  print a2;
  print a3;
  print a4;
  print a5;
}
