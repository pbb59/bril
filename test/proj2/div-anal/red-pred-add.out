main {
  zero: int = const 0;
  one: int = const 1;
  two: int = const 2;
  three: int = const 3;
  a0: int = const 7;
  a1: int = const 11;
  vec0_s: int = id a0;
  vec1_s: int = id a1;
  vec2: vector = s2v zero one two three;
  vec3: vector = s2v one zero two three;
  p0: pred = vcmp vec2 vec3;
  vec4_s: int = add vec0_s vec0_s;
  vec5_s: int = add vec1_s vec1_s;
  vec6_s: int = add vec5_s vec1_s;
  vec6_s_v: vector = s2vb vec6_s;
  vec4_s_v: vector = s2vb vec4_s;
  vec7: vector = vphi p0 vec4_s_v vec6_s_v;
  a2: int = v2s vec7 zero;
  a3: int = v2s vec7 one;
  a4: int = v2s vec7 two;
  a5: int = v2s vec7 three;
  print a2;
  print a3;
  print a4;
  print a5;
}
33
33
14
14
33
33
14
14
